### PR TITLE
refactor: use vaadin-email-field for email fields

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Replaced `vaadin-text-field` with `vaadin-email-field` in some cases where email address is expected.

### CODE REVIEW

File: frontend/demo/component/button/button-dialog.ts [hunk 1]
```diff
@@ -26,11 +26,11 @@ export class Example extends LitElement {
         <vaadin-form-layout .responsiveSteps="${[{ columns: 2 }]}">
           <vaadin-text-field label="First name" value="John"></vaadin-text-field>
           <vaadin-text-field label="Last name" value="Smith"></vaadin-text-field>
-          <vaadin-text-field
+          <vaadin-email-field
```
Comment (web-padawan) [inline frontend/demo/component/button/button-dialog.ts:29]: Java example already uses `EmailField`, so its import is already added.

File: frontend/demo/component/button/button-form.ts [hunk 2]
```diff
@@ -26,11 +26,11 @@ export class Example extends LitElement {
         <vaadin-form-layout .responsiveSteps="${[{ columns: 2 }]}">
           <vaadin-text-field label="First name" value="John"></vaadin-text-field>
           <vaadin-text-field label="Last name" value="Smith"></vaadin-text-field>
-          <vaadin-text-field
+          <vaadin-email-field
```
Comment (web-padawan) [inline frontend/demo/component/button/button-form.ts:29]: Java example already uses `EmailField`, so its import is already added.

File: frontend/demo/component/button/button-labels.ts [hunk 3]
```diff
@@ -38,13 +37,13 @@ export class Example extends LitElement {
       <!-- tag::snippet[] -->
       <vaadin-vertical-layout>
         <vaadin-horizontal-layout theme="spacing">
-          <vaadin-text-field
+          <vaadin-email-field
```
Comment (web-padawan) [inline frontend/demo/component/button/button-labels.ts:40]: Java example already uses `EmailField`, so its import is already added.

File: src/main/java/com/vaadin/demo/component/dialog/DialogHeader.java [hunk 4]
```diff
@@ -41,7 +42,7 @@ private VerticalLayout createDialogLayout(Dialog dialog) {
         nameField.setReadOnly(true);
         nameField.getStyle().set("padding-top", "0");
 
-        TextField emailField = new TextField("Email", user.getEmail(), "email@company.com");
+        EmailField emailField = new EmailField("Email", user.getEmail(), "email@company.com");
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/dialog/DialogHeader.java:45]: This constructor doesn't exist

Screenshot 2022-09-23 at 14 33 45

Comment (web-padawan) [reply src/main/java/com/vaadin/demo/component/dialog/DialogHeader.java]: Thanks, fixed.

**Review by web-padawan — COMMENTED** [review]
There are some compilation failures that I'm unable to reproduce in other examples:

```
[ERROR] /Users/serhii/vaadin/docs/src/main/java/com/vaadin/demo/component/grid/GridItemDetailsToggle.java: setLabel(java.lang.String) in com.vaadin.flow.component.textfield.GeneratedVaadinTextField cannot implement setLabel(java.lang.String) in com.vaadin.flow.component.HasLabel
[ERROR]   attempting to assign weaker access privileges; was public
[ERROR] /Users/serhii/vaadin/docs/src/main/java/com/vaadin/demo/component/grid/GridItemDetails.java: setLabel(java.lang.String) in com.vaadin.flow.component.textfield.GeneratedVaadinTextField cannot implement setLabel(java.lang.String) in com.vaadin.flow.component.HasLabel
[ERROR]   attempting to assign weaker access privileges; was public
```

**Review by web-padawan — COMMENTED** [review]
Reverted changes to Grid item details demos for now to make the build pass.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**